ionc/ion_timestamp.c

```diff
@@ -1305,6 +1308,9 @@ iERR ion_timestamp_binary_read(ION_STREAM *stream, int32_t len, decContext *cont
     // now we set the "has timezone" bit
     if (has_offset) {
         SET_FLAG_ON(ptime->precision, ION_TT_BIT_TZ);
+        ptime->tz_offset = (int16_t)-offset;
+        IONCHECK(_ion_timestamp_to_utc(ptime, ptime));
+        ptime->tz_offset = (int16_t)offset;
```

Comment:
ic, thanks for explaining.  i do wonder whether we could remove line 1215, as it doesn't appear to be adding any value (tho maybe i'm missing something...).